### ISSUE TITLE
Fix libz on Haiku

### ISSUE THREAD

**Issue by nielx** [nielx/rust#9]
There is an issue with linking libz-sys into the cargo binary during the build process. The linker complains about a hidden symbol that is referenced by a DSO (dynamic shared object). I don't know what the issue is, or why this is specifically an issue for Haiku. We solve it by having the build script of libz-sys expose all symbols.

Currently we have to comment out line 103 from build.rs, so that it looks:

```rust
    if !target.contains("windows") {
        cfg.define("STDC", None);
        cfg.define("_LARGEFILE64_SOURCE", None);
        cfg.define("_POSIX_SOURCE", None);
 //       cfg.flag("-fvisibility=hidden");
    }
```

We need to figure out why this is a problem on Haiku, or make it permanent in the libz module.

**Comment by nielx** [comment]
Could be related to https://dev.haiku-os.org/ticket/14531

**Comment by nielx** [comment]
Relevant section:
```
(...)ld: /mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/cargo-12d0297928c2f43c: hidden symbol `inflateInit2_' in /mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/liblibz_sys-6c1a1219eabdf974.rlib(inflate.o) is referenced by DSO```
```

Full command (when compiling cargo):

```
Error: linking with `x86_64-unknown-haiku-gcc` failed: exit code: 1
  |
  = note: "x86_64-unknown-haiku-gcc" "-m64" "-Wl,--eh-frame-hdr" "-L" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-haiku/lib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/cargo-12d0297928c2f43c.cargo.d47bxgq0-cgu.0.rcgu.o" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/cargo-12d0297928c2f43c.cargo.d47bxgq0-cgu.1.rcgu.o" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/cargo-12d0297928c2f43c.cargo.d47bxgq0-cgu.10.rcgu.o" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/cargo-12d0297928c2f43c.cargo.d47bxgq0-cgu.11.rcgu.o" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/cargo-12d0297928c2f43c.cargo.d47bxgq0-cgu.12.rcgu.o" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/cargo-12d0297928c2f43c.cargo.d47bxgq0-cgu.13.rcgu.o" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/cargo-12d0297928c2f43c.cargo.d47bxgq0-cgu.14.rcgu.o" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/cargo-12d0297928c2f43c.cargo.d47bxgq0-cgu.15.rcgu.o" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/cargo-12d0297928c2f43c.cargo.d47bxgq0-cgu.2.rcgu.o" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/cargo-12d0297928c2f43c.cargo.d47bxgq0-cgu.3.rcgu.o" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/cargo-12d0297928c2f43c.cargo.d47bxgq0-cgu.4.rcgu.o" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/cargo-12d0297928c2f43c.cargo.d47bxgq0-cgu.5.rcgu.o" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/cargo-12d0297928c2f43c.cargo.d47bxgq0-cgu.6.rcgu.o" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/cargo-12d0297928c2f43c.cargo.d47bxgq0-cgu.7.rcgu.o" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/cargo-12d0297928c2f43c.cargo.d47bxgq0-cgu.8.rcgu.o" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/cargo-12d0297928c2f43c.cargo.d47bxgq0-cgu.9.rcgu.o" "-o" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/cargo-12d0297928c2f43c" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/cargo-12d0297928c2f43c.4jtg7a1q2m33bxjv.rcgu.o" "-Wl,--gc-sections" "-pie" "-Wl,-zrelro" "-Wl,-znow" "-Wl,-O1" "-nodefaultlibs" "-L" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps" "-L" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/release/deps" "-L" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/build/libnghttp2-sys-7af910d1081e4ae0/out/i/lib" "-L" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/build/libz-sys-d4fd8ada3467ffa7/out/build" "-L" "/mnt/d/Haiku/system-r1beta2-x86_64/develop/lib/" "-L" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/build/libgit2-sys-f11eb75e98bdf181/out/build" "-L" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/build/libssh2-sys-84bd8615a0c22347/out/build" "-L" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-haiku/lib" "-Wl,-Bstatic" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libgit2_curl-2d6c1ec943776665.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libenv_logger-b135925695fd7bfd.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libhumantime-27250f5f01e65a5a.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libquick_error-8ff193b90c5e1d67.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libcargo-c6295bbe3141335f.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libunicode_xid-5f3fbcc2fcd8efd0.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libhex-b40af9225f12031c.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libhome-e58b4cd331916998.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libopener-265261bbe962ae1d.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libim_rc-de33951cf00035c0.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/librand_xoshiro-83b1984620394847.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libsized_chunks-1d4849e0b99d74f1.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libbitmaps-152cb8a7cb295178.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libtypenum-19e837741f21de83.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libtoml-40dffb7da1ec95dd.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libstrip_ansi_escapes-f71469a9c92b7d91.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libvte-56ebfc56d6d49a42.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libutf8parse-21b91ae6f511c668.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libhumantime-3cdc8fda1e159e1f.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libnum_cpus-674611d8e1520e70.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libserde_ignored-a684906d1810fee5.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libcrypto_hash-4f71c88cb07e3f5c.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libopenssl-4a233e6a65417d41.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libforeign_types-3bfad783d9cf4ffe.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libforeign_types_shared-80a7bcf7ecc339d0.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libhex-8b7c1e31dbbcb5e3.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libshell_escape-9cc010a64a4e1c1f.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libclap-e4417c3511b99fc3.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libvec_map-842f28a191ff9c0d.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libtextwrap-26323c77841c5039.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libunicode_width-16e8358d478fe4ad.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libstrsim-19ebc58034184a81.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libatty-82f1246c4082f217.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libansi_term-3b1e1201ed4af353.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libignore-dd5b1d5ba86cfd37.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libwalkdir-1482d00e1626d5f2.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libsame_file-6a59d1991bdf006d.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libglobset-cb871aed988f7bf0.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libregex-1628976ca5b6029a.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libthread_local-41349b8281dd486d.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libregex_syntax-545d324b77bb258e.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libfnv-2b3f3b0e79936723.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libbstr-dbef8bc60afb9ec3.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libaho_corasick-b4f4bcd7aac51c0f.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libmemchr-f113e729e032b7ce.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libcrates_io-82fd4a23819bd1e2.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/librustfix-2f2969ca37706021.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libserde_json-98a8df3018d04864.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libryu-779074e3eae35c17.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libitoa-ccb7ed380a581ac4.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libtar-cd7039bb3e143511.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libflate2-84fa90e5d566f7fc.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libcrc32fast-353ef78ded98352e.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libgit2-8a68ebc79a02a044.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libopenssl_probe-061089be6b99576e.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/liblibgit2_sys-3ac97f90e562ae5b.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/liblibssh2_sys-d036581d53189d3f.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libopenssl_sys-8a982fae78bcdd02.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/liblibz_sys-6c1a1219eabdf974.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libbitflags-11f3cce589129db5.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libtempfile-a546b16c74d22466.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/librand-169e9697b927b17a.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/librand_chacha-239f0986db971414.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libppv_lite86-3d2224a7bdaa2c5d.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/librand_core-a210d482db850f25.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libgetrandom-a6f835c8bec63c85.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libremove_dir_all-2be0b07ce0000ee1.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libglob-7c1b90a05876140e.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libtermcolor-65875034dd001019.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libcurl-1dad292a5d4414ca.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libsocket2-4e2948ace34f38d2.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libcurl_sys-c530c73b26b2fd80.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libbytesize-8f020fc4a4e2a81b.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/liburl-35b79d2b6aa3726d.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libpercent_encoding-47f4de6ab251dbf7.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libidna-286accb3a69cf8b4.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libunicode_normalization-98af789afba9a511.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libtinyvec-971c5dcbf3742401.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libunicode_bidi-50130a3bba340d8e.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libmatches-67af28b0bf01ab48.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libcrossbeam_utils-6a3acf784883a1e4.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/liblazy_static-018158dae72fd156.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/liblazycell-c8e3ee11b33508b0.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libjobserver-efbd7de0c8e40d8c.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libfiletime-8b46f3af407e670b.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/liblibc-766758effe72f6ef.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libsemver-ecf5f99c9c1da05b.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libsemver_parser-fe62700cc3ebab59.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libcargo_platform-b941be59ff2ae413.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libserde-099c0a166aefd2ab.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/liblog-fe43fb2a64bedbf3.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libcfg_if-b9ffcefeffdd9185.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/libanyhow-c0e8590e8abba1f1.rlib" "-Wl,--start-group" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-haiku/lib/libstd-269b9c7cf813a015.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-haiku/lib/libpanic_unwind-0c41b462572e6d15.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-haiku/lib/libminiz_oxide-706d2ebf0ec7739c.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-haiku/lib/libadler-1c30d1ef6f8ae57a.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-haiku/lib/libobject-793570e5bcd3de2a.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-haiku/lib/libaddr2line-72bfc62ade3b6dce.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-haiku/lib/libgimli-137e15dc455d4228.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-haiku/lib/librustc_demangle-9d7034bebffb8a37.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-haiku/lib/libhashbrown-05d6078ede14411b.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-haiku/lib/librustc_std_workspace_alloc-9c31eb3818ef5a9d.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-haiku/lib/libunwind-656218f2cc29441f.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-haiku/lib/libcfg_if-5e66af6aa2376bb1.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-haiku/lib/liblibc-421fb334360e05b5.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-haiku/lib/liballoc-0f0bff0723f4606a.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-haiku/lib/librustc_std_workspace_core-05004058973e04d5.rlib" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-haiku/lib/libcore-681288b8fbc09c5c.rlib" "-Wl,--end-group" "/mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-haiku/lib/libcompiler_builtins-8c289de990425963.rlib" "-Wl,-Bdynamic" "-lssl" "-lcrypto" "-lcurl" "-lbsd" "-lgcc_s" "-lroot" "-lnetwork" "-lbsd"
  = note: /mnt/d/Haiku/build-r1beta2-x86_64/cross-tools-x86_64/lib/gcc/x86_64-unknown-haiku/8.3.0/../../../../x86_64-unknown-haiku/bin/ld: /mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/cargo-12d0297928c2f43c: hidden symbol `inflateInit2_' in /mnt/d/Rust/rust-x86/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-haiku/release/deps/liblibz_sys-6c1a1219eabdf974.rlib(inflate.o) is referenced by DSO
          /mnt/d/Haiku/build-r1beta2-x86_64/cross-tools-x86_64/lib/gcc/x86_64-unknown-haiku/8.3.0/../../../../x86_64-unknown-haiku/bin/ld: final link failed: bad value
          collect2: error: ld returned 1 exit status
```

**Comment by nielx** [comment]
The issue actually seems to be that Haiku's libcurl.so depends on libz.so. This should be made explicit in the curl-sys build system.

**Comment by nielx** [comment]
Unfortunately adding -lz to libcurl.so  does not make the warning disappear.

**Comment by nielx** [comment]
PR submitted: https://github.com/rust-lang/libz-sys/pull/75

**Comment by nielx** [comment]
Included in libz 1.1.3
